feature_engine/transformation/arcsin.py

```diff
@@ -0,0 +1,178 @@
+# Authors: Tommaso Pellegrino <tommasopellegrino.1995@gmail.com>
+# License: BSD 3 clause
+
+from typing import List, Optional, Union
+
+import numpy as np
+import pandas as pd
+
+from feature_engine._docstrings.class_inputs import _variables_numerical_docstring
+from feature_engine._docstrings.fit_attributes import (
+    _feature_names_in_docstring,
+    _n_features_in_docstring,
+    _variables_attribute_docstring,
+)
+from feature_engine._docstrings.methods import (
+    _fit_not_learn_docstring,
+    _fit_transform_docstring,
+    _inverse_transform_docstring,
+)
+from feature_engine._docstrings.substitute import Substitution
+from feature_engine.base_transformers import BaseNumericalTransformer
+from feature_engine.tags import _return_tags
+from feature_engine.variable_manipulation import _check_input_parameter_variables
+
+
+@Substitution(
+    variables=_variables_numerical_docstring,
+    variables_=_variables_attribute_docstring,
+    feature_names_in_=_feature_names_in_docstring,
+    n_features_in_=_n_features_in_docstring,
+    fit=_fit_not_learn_docstring,
+    fit_transform=_fit_transform_docstring,
+    inverse_transform=_inverse_transform_docstring,
+)
+class ArcsinTransformer(BaseNumericalTransformer):
+    """
+    The ArcsinTransformer() applies the arcsin transformation to numerical variables.
+
```

Comment:
Can we add the following paragraphs here:

The arcsine transformation, also called arcsin square root transformation, or
angular transformation, takes the form of asin(sqrt(x)) where x is a real number
between 0 and 1.

The arcsin square root transformation helps in dealing with probabilities,
percents, and proportions.

Reply:
done